frontend/appflowy_flutter/lib/plugins/document/presentation/editor_plugins/actions/keys_pressed.dart

```diff
@@ -0,0 +1,63 @@
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+
+class KeysPressed extends StatefulWidget {
+  final Widget child;
+
+  const KeysPressed({
+    super.key,
+    required this.child,
+  });
+
+  @override
+  State<KeysPressed> createState() => _KeysPressedState();
+}
+
+class _KeysPressedState extends State<KeysPressed> {
+  final _focusNode = FocusNode();
+
+  bool _isAltPressed = false;
+
+  @override
+  Widget build(BuildContext context) {
+    return KeyboardListener(
+      focusNode: _focusNode,
+      onKeyEvent: (event) {
+        if (event is! KeyRepeatEvent) {
+          return;
+        }
+
+        /// TODO: Improve to allow for any key we need to track
+        if (LogicalKeyboardKey.altLeft == event.logicalKey) {
+          final isPressed = event is KeyDownEvent;
+
+          if (isPressed != _isAltPressed) {
+            setState(() => _isAltPressed = isPressed);
+          }
+        }
+      },
+      child: KeysPressedManager(
+        isAltPressed: _isAltPressed,
+        child: widget.child,
+      ),
+    );
+  }
+}
+
+class KeysPressedManager extends InheritedWidget {
+  final bool isAltPressed;
```

Comment:
We're probably going to extend this later down the line and start adding a `isPressed` for many keyboard keys. I wonder if there's a way that we can create a hash table/lookup instead of adding a field for each key that we want to check if it is pressed.

Reply:
Yeah, I'll expand on this down the road, hence the TODO. :+1: